builtin.c

```diff
@@ -197,6 +197,55 @@ static OP *ck_builtin_func1(pTHX_ OP *entersubop, GV *namegv, SV *ckobj)
     return newUNOP(opcode, wantflags, argop);
 }
 
+XS(XS_builtin_indexed)
+{
+    dXSARGS;
+
+    switch(GIMME_V) {
+        case G_VOID:
+            Perl_ck_warner(aTHX_ packWARN(WARN_VOID),
+                "Useless use of builtin::indexed in void context");
+            XSRETURN(0);
+            break;
+        case G_SCALAR:
+            Perl_ck_warner(aTHX_ packWARN(WARN_SCALAR),
+                "Useless use of builtin::indexed in scalar context");
```

Comment:
> The existing Useless use of sort in scalar context warning is in the void category. Should that be changed?

Possibly. Actually I wonder about splitting that out into its own PR; adding a "scalar" category and moving the existing one first, then rebase this PR atop that.

> Also, the perldiag entry should be updated to Useless use of %s in scalar context and the wording adjusted accordingly.

I expected it would but it didn't fail for me locally so I left it alone. It seems the smokers have found it though.

Reply:
> > The existing Useless use of sort in scalar context warning is in the void category. Should that be changed?
> 
> Possibly. Actually I wonder about splitting that out into its own PR; adding a "scalar" category and moving the existing one first, then rebase this PR atop that.

Done https://github.com/Perl/perl5/pull/19510